controller/lib/debug/debug.cpp

```diff
@@ -0,0 +1,336 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * The debug serial port interface implements a very simple binary command
+ * structure.  Commands are sent using the following format:
+ *
+ *   <cmd> <data> <crc> <term>
```

Comment:
It certainly could be.  I actually suggested this exact type of framing in a comment on the #159 issue.
I haven't really been paying too much attention to that particular issue, I figured it would have been resolved by now.  If it's still an open issue then using this approach makes sense.

Reply:
What are your thoughts on how to make sure we don't duplicate with #322 which also implements framing? We could do just "whoever merges first, wins; the other author then should use the first library (and extract a library if it hasn't been extracted yet)" but that could lead to some wasted effort.